activation/post_states.go

```diff
@@ -0,0 +1,103 @@
+package activation
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	pb "github.com/spacemeshos/api/release/go/spacemesh/v1"
+	"go.uber.org/zap"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/events"
+)
+
+type postStates struct {
+	log            *zap.Logger
+	mu             sync.RWMutex
+	states         map[types.NodeID]postState
+	watchingStates sync.Once
+}
+
+func newPostStates(log *zap.Logger) postStates {
+	return postStates{
+		log:    log,
+		states: make(map[types.NodeID]postState),
+	}
+}
+
+type postState int
```

Comment:
I think this can be made a public type then `activation.PostState` can be used in the mapping to `pb.PostState_State` in the GRPC service -> makes it less likely to forget to update it when we change / extend this enum.